packages/palette/src/elements/Modal/index.tsx

```diff
@@ -0,0 +1,238 @@
+import { Flex } from "../Flex"
+import { Spacer } from "../Spacer"
+import { Serif } from "../Typography"
+
+interface ModalProps {
+  title?: string
+  isWide?: boolean
+  hasLogo?: boolean
+  FixedButton?: JSX.Element
+  show?: boolean
+  onClose?: () => void
+}
+
+interface TransitionElementProps {
+  show?: boolean
+  isAnimatingOpen?: boolean
+  isAnimatingClosed?: boolean
+  isWide?: boolean
+}
+
+const usePrevious = value => {
+  const ref = useRef()
+  useEffect(() => {
+    ref.current = value
+  }, [value])
+  return ref.current
+}
+
+const AnimatedView = animated(Box)
+
+/**
+ * Modal.
+ * Spec: https://app.zeplin.io/project/5acd19ff49a1429169c3128b/screen/5c75ad115c1db5628cc03c2a
+ */
+export const Modal: SFC<ModalProps> = ({
+  children,
+  FixedButton,
+  title,
+  show,
+  isWide,
+  hasLogo,
+  onClose,
+}) => {
+  const [childrenState, setChildrenState] = useState(children)
+  const [titleState, setTitleState] = useState(title)
+  const [fixedButtonState, setFixedButtonState] = useState(FixedButton)
+  const [hasLogoState, setHasLogoState] = useState(hasLogo)
+  const [springContentAnimation, setSpringContentAnimation] = useState({
+    opacity: 1,
+    onRest: null,
+  })
+  const [springModalAnimation, setSpringModalAnimation] = useState({
+    transform: "translate(-50%, 0%) translateY(50vh)",
+    opacity: 1,
+    onRest: null,
+  })
+  const [visibleContent, setVisibleContent] = useState(null)
+
+  const previousChildren = usePrevious(childrenState)
+  const previousFixedButton = usePrevious(fixedButtonState)
+  const previousHasLogo = usePrevious(hasLogoState)
+  const previousTitle = usePrevious(titleState)
+  const contentAnimation = useSpring(springContentAnimation)
+  const modalAnimation = useSpring(springModalAnimation)
+
+  useEffect(() => {
+    if (!show) {
+      setSpringModalAnimation({
+        transform: "translate(-50%, -50%) translateY(0vh)",
+        opacity: 0,
+        onRest: () => {
+          setSpringModalAnimation({
+            transform: "translate(-50%, 0%) translateY(50vh)",
+            onRest: null,
+            opacity: 0,
+          })
+          onClose()
+        },
+      })
+    } else {
+      setVisibleContent(content)
+      setSpringModalAnimation({
+        transform: "translate(-50%, -50%) translateY(0vh)",
+        opacity: 1,
+        onRest: null,
+      })
+    }
+  }, [show])
+
+  // Listens for props to change and fades in new content
+  useEffect(() => {
+    if (
+      (previousChildren && children !== previousChildren) ||
+      (previousFixedButton &&
+        FixedButton &&
+        FixedButton.props.children !== previousFixedButton.props.children) ||
+      (previousTitle && title !== previousTitle) ||
+      (previousHasLogo && hasLogo !== previousHasLogo)
+    ) {
+      setSpringContentAnimation({
+        opacity: 0,
+        onRest: () => updateVisibleContent(),
+      })
+    }
+  }, [hasLogo, title, FixedButton, children])
+
+  const content = () => {
```

Comment:
#minor - this is technically a FC component, so should be `const SomeComponent = () => { ... }` (caps case)

Reply:
Fixed this